skimage/io/collection.py

```diff
@@ -124,76 +125,42 @@ class ImageCollection:
-    ``load_func(load_pattern[5])`` to load the image.
-
-    Imagine, for example, an ImageCollection that loads every third
-    frame from a video file::
-
-      video_file = 'no_time_for_that_tiny.gif'
-
-      def vidread_step(f, step):
-          vid = imageio.get_reader(f)
-          seq = [v for v in vid.iter_data()]
-          return seq[::step]
-
-      ic = ImageCollection(video_file, load_func=vidread_step, step=3)
-
-      ic  # is an ImageCollection object of length 1 because there is 1 file
-
-      x = ic[0]  # calls vidread_step(video_file, step=3)
-      x[5]  # is the sixth element of a list of length 8 (24 / 3)
-
-    Alternatively, if `load_func` is provided and `load_pattern` is a
-    sequence, an `ImageCollection` of corresponding length will be created,
-    and the individual images will be loaded by calling `load_func` with the
-    matching element of the `load_pattern` as its first argument. In this
-    case, the elements of the sequence do not need to be names of existing
-    files (or strings at all). For example, to create an `ImageCollection`
-    containing 500 images from a video::
-
-      class vidread_random:
-          def __init__ (self, f):
-              self.vid = imageio.get_reader(f)
-          def __call__ (self, frameno):
-              return self.vid.get_data(frameno)
-      ic = ImageCollection(range(500), load_func=vidread_random('movie.mp4'))
-
-      ic  # is an ImageCollection object of length 500
-
-    Another use of `load_func` would be to convert all images to ``uint8``::
-
-      def imread_convert(f):
-          return imread(f).astype(np.uint8)
-
-      ic = ImageCollection('/tmp/*.png', load_func=imread_convert)
-
     Examples
     --------
-    >>> import imageio
-    >>> import skimage.io as io
-    >>> from skimage import data_dir
-
-    >>> coll = io.ImageCollection(data_dir + '/chess*.png')
-    >>> len(coll)
+    >>> import skimage as ski
+    >>> collection = ski.io.ImageCollection(ski.data.data_dir + '/chess*.png')
+    >>> len(collection)
     2
-    >>> coll[0].shape
+    >>> collection[0].shape
     (200, 200)
 
-    >>> image_col = io.ImageCollection(['/tmp/work/*.png', '/tmp/other/*.jpg'])
-
-    >>> class multiread:
-    ...     def __init__ (self, f):
-    ...         self.vid = imageio.get_reader(f)
-    ...     def __call__ (self, frameno):
-    ...         return self.vid.get_data(frameno)
+    If `load_func` is provided and `load_pattern` is a sequence of objects other than
+    strings, an `ImageCollection` of corresponding length will be created, and the
+    individual images will be loaded by calling `load_func` with the matching element
+    of the `load_pattern` as its first argument. In this case, the elements of the
+    sequence do not need to be names of existing files (or strings at all). For example,
+    to create an `ImageCollection` that of frames based on an increasing counter with a
+    specific width:
```

Comment:
A counter cannot have a width, so what does this mean?

Reply:
True, that didn't make much sense. I tried to improve it in https://github.com/scikit-image/scikit-image/pull/6934/commits/73133601d808ecab464335b81cf06f0696d83025.